packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -463,6 +463,19 @@ export class ProfilesUtils {
             if (typeof profile !== "string") {
                 await Constants.PROFILES_CACHE.updateCachedProfile(profile, node);
             }
+            if (node !== null) {
+                const toolTipList = (node.tooltip as string)?.split("\n") ?? [];
+                const userIDIndex = toolTipList.findIndex((key) => key.startsWith(vscode.l10n.t("User: ")));
+                if (userIDIndex !== -1) {
+                    toolTipList[userIDIndex] = `${vscode.l10n.t("User: ")}${creds[0]}`;
+                } else {
+                    toolTipList.push(`${vscode.l10n.t("User: ")}${creds[0]}`);
+                }
+                node.tooltip = toolTipList.join("\n");
+                if (node) {
+                    void node.getChildren().then(() => SharedTreeProviders.getProviderForNode(node).refreshElement(node));
```

Comment:
I believe this logic is to implement a refresh after changing the username. I would suggest using the `refreshElement` function instead to avoid iterating over all the children and a possible duplicate refresh:

```suggestion
                    SharedTreeProviders.getProviderForNode(node).refreshElement(node);
```

Reply:
Addressed it in the following commit @traeok .